captum/attr/_utils/gradient.py

```diff
@@ -372,7 +385,15 @@ def compute_layer_gradients_and_eval(
         # key_list is a list of devices in appropriate ordering for concatenation.
         # If only one key exists (standard model), key list simply has one element.
         key_list = _sort_key_list(list(saved_layer.keys()), device_ids)
-        all_outputs = _reduce_list([saved_layer[device_id] for device_id in key_list])
+
+        all_outputs = _reduce_list(
+            [
+                saved_layer[device_id]
+                if output_fn is None
+                else output_fn(saved_layer[device_id])
```

Comment:
Since we were already iterating in the loop, I added it in the loop: https://github.com/pytorch/captum/blob/master/captum/attr/_utils/gradient.py#L179